utils/tinystr/src/unvalidated.rs

```diff
@@ -0,0 +1,88 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::TinyAsciiStr;
+use crate::TinyStrError;
+
+#[derive(Debug, PartialEq, PartialOrd, Eq, Ord, Clone, Copy)]
+pub struct UnvalidatedTinyAsciiStr<const N: usize>(pub(crate) [u8; N]);
+
+impl<const N: usize> UnvalidatedTinyAsciiStr<N> {
+    pub fn try_into_tinystr(&self) -> Result<TinyAsciiStr<N>, TinyStrError> {
+        TinyAsciiStr::try_from_raw(self.0)
+    }
+}
+
+impl<const N: usize> TinyAsciiStr<N> {
+    pub fn raw(self) -> UnvalidatedTinyAsciiStr<N> {
```

Comment:
Discussion: not sure if `raw` is the right name for this function

Reply:
`to_unvalidated()` imo